TombLib/TombLib/LevelData/Compilers/Util/TrxInjector.cs

```diff
@@ -0,0 +1,174 @@
+﻿using ICSharpCode.SharpZipLib.Zip.Compression.Streams;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using TombLib.IO;
+
+namespace TombLib.LevelData.Compilers.Util;
+
+public static class TrxInjector
+{
+    private const uint _magic = 'T' | 'R' << 8 | 'X' << 16 | 'J' << 24;
+    private const uint _version = 2;
+    private const uint _injectionType = 0; // Implies no link to a TRX config option
+
+    public static void Serialize(TrxInjectionData data, BinaryWriterEx outWriter)
+    {
+        using MemoryStream stream = new();
+        using BinaryWriterEx injWriter = new(stream);
+
+        if (!WriteData(data, injWriter))
+        {
+            return;
+        }
+
+        var exportedData = stream.ToArray();
+        using var outStream = new MemoryStream();
+        using var deflater = new DeflaterOutputStream(outStream);
```

Comment:
We already have zlib integrated in TE codebase. Consider using zlib compression instead of zip.

Reply:
Fixed, thanks.